lib/ansible/modules/cloud/azure/azure_rm_autoscale.py

```diff
@@ -0,0 +1,620 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_autoscale
+version_added: "2.7"
+short_description: Manage Azure image.
+description:
+    - Create, delete an image from virtual machine, blob uri, managed disk or snapshot.
+options:
+    target:
+        description:
+        - The resource identifier of the resource that the autoscale setting should be added to.
+        - It can be the resource id of the resource.
```

Comment:
I would write: "it could be resource id string or a dictionary containing C(name)....."